code/atom.dm

```diff
@@ -158,6 +158,11 @@
 	proc/Scale(var/scalex = 1, var/scaley = 1)
 		src.transform = matrix(src.transform, scalex, scaley, MATRIX_SCALE)
 
+	// a turn-safe scale, for temporary anisotropic scales
+	proc/SafeScale(var/scalex = 1, var/scaley = 1)
+		var/rot = arctan(src.transform.b, src.transform.a)
+		src.transform = matrix(matrix(matrix(src.transform, -rot, MATRIX_ROTATE), scalex, scaley, MATRIX_SCALE), rot, MATRIX_ROTATE)
```

Comment:
What is this doing differently than Scale?

Reply:
It sets the current rotation to zero before it applies the scale. The problem in #2474 was that a rotation was being applied to the atom between the initial and final calls to scale (which I will call scale and de-scale), so what was intended to be a de-scaling did not correctly apply the de-scaling to the rotated atom. That is, 

```
atom.Scale(A, B)
SPAWN_DBG(N)
atom.Scale(C, D)
```

is not guaranteed to be equivalent to `atom.Scale(A * C, B * D)`, as a rotation could happen between the two calls to `Scale`. 

By negating the current rotation, applying the scaling, and then restoring the rotation, the following two blocks of code are guaranteed to result in equivalent changes to `src.transform`:

```
atom.SafeScale(A, B)
SPAWN_DBG(N)
atom.SafeScale(C, D)
```

```
atom.SafeScale(A * C, B * D)
```

I haven't tested this with shear or translations, but since the problem here was with rotating a sprite while Bubsium is in its system, I figured this was good enough.